src/main.rs

```diff
@@ -138,7 +139,7 @@ fn show_add_input(mut todos: &mut Vec<Todo>, screen: &mut i8, window: &WINDOW) {
     curs_set(CURSOR_VISIBILITY::CURSOR_INVISIBLE); // Don't show the terminal cursor
 }
 
-fn list_todos(todos: &Vec<Todo>, cur_index: i32) {
+fn list_todos(todos: &[Todo], cur_index: i32) {
```

Comment:
Can you explain this syntax to me? Is is vector specific?

Reply:
This came up when I ran `cargo clippy`. To see the output exactly, checkout to my first commit in this PR, and run `cargo install clippy && cargo clippy`.

It was the first time I came across this myself. Looking into it, if your function doesn't change the size of your vector, you want to pass in a reference to the [inner-array](https://github.com/rust-lang/rust/blob/4bd32c98047a809ba5fd1fac2aa044638e5f2105/src/liballoc/raw_vec.rs#L47)(note, `ptr: Unique<T>` is effectively `ptr: &[T]` in this context), rather than a reference to the [struct itself](file:///home/ben/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/share/doc/rust/html/src/alloc/vec.rs.html#300-303). Note how a `Vec` is just a buffer and a size. If that size is unchanged, then the abstractions wrapping the buffer can be thrown away.